sdk/core/System.ClientModel/tests/Pipeline/ClientLoggingPolicyTests.cs

```diff
@@ -0,0 +1,292 @@
+        Assert.AreEqual("sample-client-B", responseLog.GetProperty<string>("requestId"));
+    }
+
+    [Test]
+    public async Task RandomGuidIsSetForRequestIdWhenNotProvided()
+    {
+        using ClientCustomEventListener clientEventListener = new();
+
+        MockPipelineResponse response = new(200, mockHeaders: new MockResponseHeaders());
+        response.SetContent("Response");
+        ClientPipelineOptions options = new()
+        {
+            Transport = new MockPipelineTransport("Transport", i => response),
+            LoggingPolicy = new ClientCustomLoggingPolicy(new LoggingOptions() { LoggedClientAssemblyName = "SampleSDK" })
+        };
+        ClientPipeline pipeline = ClientPipeline.Create(options);
+
+        PipelineMessage message = pipeline.CreateMessage();
+        message.Request.Uri = new Uri("http://example.com");
+        message.Request.Content = BinaryContent.Create(new BinaryData("Request"));
+
+        await pipeline.SendSyncOrAsync(message, IsAsync);
+
+        EventWrittenEventArgs requestLog = clientEventListener.SingleEventById(RequestEvent);
+        Guid requestId = Guid.Parse(requestLog.GetProperty<string>("requestId"));
+
+        EventWrittenEventArgs responseLog = clientEventListener.SingleEventById(ResponseEvent);
+        Guid responseId = Guid.Parse(responseLog.GetProperty<string>("requestId"));
+
+        Assert.AreEqual(requestId, responseId);
+    }
+
+    [Test]
+    public async Task RequestIdIsUsedWhenNoResponseIdIsInHeaders()
+    {
+        using ClientCustomEventListener clientEventListener = new();
+        string clientIdHeaderName = "Client-ID";
+
+        MockPipelineResponse response = new(200, mockHeaders: new MockResponseHeaders());
+        response.SetContent("Response");
+        ClientPipelineOptions options = new()
+        {
+            Transport = new MockPipelineTransport("Transport", i => response),
+            LoggingPolicy = new ClientCustomLoggingPolicy(new LoggingOptions() { LoggedClientAssemblyName = "SampleSDK", RequestIdHeaderName = clientIdHeaderName })
+        };
+        ClientPipeline pipeline = ClientPipeline.Create(options);
+
+        PipelineMessage message = pipeline.CreateMessage();
+        message.Request.Uri = new Uri("http://example.com");
+        message.Request.Headers.Add(clientIdHeaderName, "sample-client");
+        message.Request.Content = BinaryContent.Create(new BinaryData("Request"));
+
+        await pipeline.SendSyncOrAsync(message, IsAsync);
+
+        EventWrittenEventArgs requestLog = clientEventListener.SingleEventById(RequestEvent);
+        Assert.AreEqual("sample-client", requestLog.GetProperty<string>("requestId"));
+
+        EventWrittenEventArgs responseLog = clientEventListener.SingleEventById(ResponseEvent);
+        Assert.AreEqual("sample-client", responseLog.GetProperty<string>("requestId"));
+    }
+
+    #region Helpers
+
+    private class AzureCoreLoggingPolicy : ClientLoggingPolicy
```

Comment:
Apologies for my memory failing me here -- remind me -- does this simulate the scenario where an Azure client replaces its Azure.Core HttpPipeline with an SCM ClientPipeline, and uses a subtype of the SCM ClientLoggingPolicy in its pipeline to implement logging without making breaking changes to the client's original logging functionality?

Reply:
No, but that's a good test that I will add. I'm just using this to test the singleton logic to make sure that multiple logging policies can be created with the same event source name, different event source names, etc and all the events are written correctly

Reply:
Although I think the event source tests enforce the non breaking changes pretty well. They check event numbers and names, argument types, etc